tracer/src/Datadog.Trace/Telemetry/Metrics/MetricTags.cs

```diff
@@ -243,6 +243,13 @@ public enum WafAnalysis
         [Description("waf_version;rule_triggered:false;request_blocked:false;waf_timeout:false;request_excluded:true")]RequestExcludedViaFilter,
     }
 
+    public enum TruncationReason
+    {
+        [Description("truncation_reason:string_too_long")]StringTooLong = 1,
+        [Description("truncation_reason:list_or_map_too_large")]ListOrMapTooLarge = 2,
+        [Description("truncation_reason:object_too_deep")]ObjectTooDeep = 4,
```

Comment:
Do you want to tag these by `waf_version` too? If so, you'll want something like this:


```suggestion
        [Description("waf_version;truncation_reason:string_too_long")]StringTooLong = 1,
        [Description("waf_version;truncation_reason:list_or_map_too_large")]ListOrMapTooLarge = 2,
        [Description("waf_version;truncation_reason:object_too_deep")]ObjectTooDeep = 4,
```

You'd need to regenerate etc though

Reply:
I don't think it makes sense, because truncation depends on the client app input, not the version of the WAF.